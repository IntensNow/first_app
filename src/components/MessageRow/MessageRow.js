import React, { PropTypes } from 'react';
import './main.less';

const MessageRow = props => {
    
    const { author, text, date } = props.msg;
    
    return (
        <div className={ author === 'Me' ? 'user_message_row ' : 'message_row' }>
            <header > <p>{author}</p> </header>
            <div className='message_text'><p>{text.trim()}</p></div>
            <div className='message_date'><p>{date}</p></div>
        </div>
    )
}

MessageRow.propTypes = {
    msg:PropTypes.shape({
        author:PropTypes.string,
        text:PropTypes.string,
        date:PropTypes.string
    })
}

export default MessageRow;
