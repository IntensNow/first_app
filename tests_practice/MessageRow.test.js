import React from 'react';
import renderer from 'react-test-renderer';
import MessageRow from '../src/components/MessageRow';

const msg = {
    author:'Me',
    text:'some text for message',
    date:'23.02.2027'
}

test('<MessageRow />', () => {
  
     const tree = renderer.create(
    
     <MessageRow msg={msg}/>
    
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    
});