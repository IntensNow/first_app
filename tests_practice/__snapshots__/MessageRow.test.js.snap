// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageRow /> 1`] = `
<div
  className="user_message_row "
>
  <header>
     
    <p>
      Me
    </p>
     
  </header>
  <div
    className="message_text"
  >
    <p>
      some text for message
    </p>
  </div>
  <div
    className="message_date"
  >
    <p>
      23.02.2027
    </p>
  </div>
</div>
`;
